x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/create_rules_bulk_route.ts

```diff
@@ -48,6 +48,35 @@ export const createCreateRulesBulkRoute = (server: ServerFacade): Hapi.ServerRou
         return headers.response().code(404);
       }
 
+      // Determine if duplicate rule_ids exist in the request
+      // before attempting to create them.
+      const mapDuplicates = request.payload.reduce<Map<string, number>>((acc, rule) => {
+        if (rule.rule_id != null && acc.has(rule.rule_id)) {
+          const totalView = acc.get(rule.rule_id) ?? 1;
+          acc.set(rule.rule_id, totalView + 1);
+        } else if (rule.rule_id != null) {
+          acc.set(rule.rule_id, 1);
+        }
+        return acc;
+      }, new Map());
```

Comment:
optional: I would pull this out into its own pure function that takes the payload and returns the mapDuplicates. It would make it easier to write a test or two directly against it and keep the controller/side effectly logic of `create_rules_bulk_route.ts` from ballooning up which is always a good goal.